magda-sleuther-format-layerer/src/format-engine/measures.ts

```diff
@@ -0,0 +1,196 @@
+* TODO not unit tested
+*/
+export class DownloadMeasure extends Measure {
+    public relatedDistribution: Record;
+    getSelectedFormats(): SelectedFormats {
+        const { downloadURL } = this.relatedDistribution.aspects[
+            "dcat-distribution-strings"
+        ];
+        const rawMime: string = mimeTypes.lookup(downloadURL);
+
+        const processedMime: string = rawMime.split(rawMime)[1];
+
+        this.canDeduceFormat = true;
+        return {
+            selectedFormats: [
+                {
+                    format: (<any>Formats)[processedMime],
+                    correctConfidenceLevel: 100,
+                    distribution: this.relatedDistribution
+                }
+            ]
+        };
+    }
+}
+
+/*
+* Tries to determine the format by deciphering DCAT-DISTRIBUTION-STRING -> format
+* TODO Unit tests
+*/
+export class DcatFormatMeasure extends Measure {
+    public relatedDistribution: Record;
+
+    private applyTransform(format: string, transformFormat: any) {
+        const newFormat = transformFormat(format);
+        return newFormat;
+    }
+
+    private empty(str: string) {
+        return "";
+    }
+
+    private lowerCase(str: string) {
+        return str.toLowerCase();
+    }
+
+    private removePeriod(str: string) {
+       return str.split('.').join('');
+    }
+
+    private replaceCommaThenSpace(str: string) {
+        return str.split(', ').join(' ');
+    }
+
+    private replaceSpaceThenComma(str: string) {
+        return str.split(' ,').join(' ');
+    }
+
+    private replaceComma(str: string) {
+        return str.split(',').join(' ');
+    }
+
+    private identity(str: string) {
+        return str;
+    }
+
+    getSelectedFormats(): SelectedFormats {
+        const { format } = this.relatedDistribution.aspects[
+            "dcat-distribution-strings"
+        ];
+
+        if(format === null || format === "") {
+            this.canDeduceFormat = false;
+            return null;
+        }
+
+        let processedFormat: string;
+
+        const transforms = [
+            this.identity,
+            (str: string) => this.lowerCase(this.removePeriod(this.replaceComma(this.replaceSpaceThenComma(this.replaceCommaThenSpace(str)))))
+        ];
+
+        for (const tf of transforms) {
+            processedFormat = this.applyTransform(format, tf);
+        }
+
+        // hard coded rules for separating out multiple formats when provided
+        let splitFormat: Array<string>;
+        let finalFormat: Array<string>;
+
+        // e.g: application/exe
+        splitFormat = processedFormat.split("/");
+        if (splitFormat.length > 2)
+            throw new Error(
+                "a MIME type has more than 1 slash: " + processedFormat
+            );
+        if (splitFormat.length > 1) finalFormat[0] = splitFormat[1];
+        else {
+            // E.g. pdf & xlsx & doc & docx
+            splitFormat = processedFormat.split(" & ");
```

Comment:
This might work better as something like `split(\/s*(&)/s*\)` in case the number of spaces varies or the separator is different.

Also isn't this pretty similar to what you're doing above with the commas, full stops etc? Might be easier to do it all in one pass.